deploy/azure/ARM-for-single-account.json

```diff
@@ -217,29 +242,11 @@
                                 ]
                             }
                         },
-                        {
-                            "type": "Microsoft.Compute/virtualMachines/extensions",
-                            "apiVersion": "2021-04-01",
-                            "name": "cloudbeatVM/EnableMSIExtension",
-                            "location": "[resourceGroup().location]",
-                            "dependsOn": [
-                                "cloudbeatVM"
-                            ],
-                            "properties": {
-                                "publisher": "Microsoft.ManagedIdentity",
-                                "type": "ManagedIdentityExtensionForLinux",
-                                "typeHandlerVersion": "1.0",
-                                "autoUpgradeMinorVersion": true,
-                                "settings": {
-                                    "port": 50342
-                                }
-                            }
-                        },
```

Comment:
Does this mean this is was not necessary before?

Reply:
It was. I discussed with @jeniawhite and it was there to grant access to the graph API.

The `Microsoft.Compute/virtualMachines/extensions` `Microsoft.ManagedIdentity` has been deprecated and should not be used. The only place I found a deprecation was on [azure cli release notes](https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli#vm-27). I don't know the reason nor could find a documentation for a replacement. But still, that is a rule in the linter.

Through reading around (and some assumptions), it seems that the 

```
"identity": {
  "type": "SystemAssigned"
}
```

Should suffice nowadays giving a managed identity to the VM.

I tested and verified everything works as expected ([tests](https://github.com/elastic/cloudbeat/issues/1663#issuecomment-2074358752))